eq-author-api/middleware/loadQuestionnaire.js

```diff
@@ -1,9 +1,21 @@
-const loadQuestionnaire = require("../utils/loadQuestionnaire");
+const { QuestionnanaireVersionsModel } = require("../db/models/DynamoDB");
 
-module.exports = (logger, context) => (req, res, next) => {
+module.exports = (logger, context) => async (req, res, next) => {
   const questionnaireId = req.header("questionnaireId");
   if (questionnaireId) {
-    context.questionnaire = JSON.parse(loadQuestionnaire(questionnaireId));
+    context.questionnaire = await new Promise((resolve, _) => {
+      QuestionnanaireVersionsModel.queryOne({ id: { eq: questionnaireId } })
+        .descending()
+        .exec((err, questionnaire) => {
+          if (err) {
+            return console.log(err);
+          }
+          if (!questionnaire.metadata) {
```

Comment:
Why are we doing this? Don't all questionnaires have metadata?

Reply:
For some reason the empty Metadata arrays were not being persisted. I need to look at this